### PR TITLE
Expand validation of nested map key type to Enum and Integer

### PR DESCRIPTION
**Pull request by FrauBoes**
In addition to String, allow Enum and Integer as key type in a nested map.

Follow-up of #23045.

### CODE REVIEW

**Comment by FrauBoes** [comment]
@bot-gradle test this

**Comment by FrauBoes** [comment]
@bot-gradle test this

**Comment by FrauBoes** [comment]
@bot-gradle test this

**Comment by FrauBoes** [comment]
@bot-gradle test this

File: subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/NestedBeanAnnotationHandler.java [hunk 1]
```diff
@@ -59,15 +61,19 @@ public void visitPropertyValue(String propertyName, PropertyValue value, Propert
 
     private static final Set<Class<?>> SUPPORTED_KEY_TYPES = new HashSet<>(Arrays.asList(String.class, Integer.class, Enum.class));
 
+    private static String getSupportedKeyTypes() {
+        return SUPPORTED_KEY_TYPES.stream().sorted(Comparator.comparing(Class::toString)).map(cls -> "'" + cls.getSimpleName() + "'").collect(Collectors.joining(", "));
+    }
```
Comment (lptr) [inline subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/NestedBeanAnnotationHandler.java:66]: 🧐 It's good that you thought of sorting the entries! It'd be even better to use a data structure that doesn't need sorting. For static collections we always use Guava's `ImmutableSet` and friends. These collections are immutable, and they retain insertion order (even without the extra overhead of a `LinkedHashSet`), so you can pick what order the elements will appear. `ImmutableCollection`s have nice static factory methods and builders that allow them to be constructed with less boilerplate, too.

**Review by lptr — APPROVED** [review]
LGTM, left a small comment.

File: subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/NestedBeanAnnotationHandler.java (hunk 1 above)
Comment (FrauBoes) [reply subprojects/model-core/src/main/java/org/gradle/internal/properties/annotations/NestedBeanAnnotationHandler.java:66]: Great, thanks for pointing that out!

**Comment by FrauBoes** [comment]
@bot-gradle test and merge

**Pull request merged**